ansible_collections/arista/avd/plugins/plugin_utils/topology_generator_utils.py

```diff
@@ -0,0 +1,715 @@
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+import os
+import queue
+import math
+
+import yaml
+import textwrap
+
+import drawSvg as draw
+
+# MARGINS between real groups
+BOXMARGIN = 25
+
+# Sizings
+ROUTERMAXFONTSIZE = 25
+ROUTERWRAPFONTSIZE = 18
+ROUTERSIZE = 120
+ROWSPACING = ROUTERSIZE * 2
+LVLSPACING = ROUTERSIZE * 2
+PORTHEIGHT = 30
+PORTWIDTH = 20
+PORTOFFSET = 3
+PORTFONTSIZE = 16
+TITLEFONTSIZE = 24
+TITLEOFFSET = 0
+FONTWIDTHHEIGHTRATIO = 0.62
+
+# Used for horizontal connection overlaps
+WIREDEFAULTDISTANCE = 10
+WIRESPREAD = 6
+
+EDGETYPES = ["server"]
+
+def read_yaml_file(filename):
+    """
+    Open a structure config file and load a data from the file
+
+    Args:
+        filename: Name of the file
+
+    Returns:
+        Data from the file
+    """
+    with open(filename, "r", encoding="utf-8") as stream:
+        data = None
+        try:
+            data = yaml.safe_load(stream)
+        except yaml.YAMLError as exc:
+            print(exc)
+        return data
+
+
+def create_node_dict(file_data, filename):
+    """
+    Create a dictionary of nodes with its corresponding neighbors
+
+    Args:
+        file_data: Data of structure config file
+        filename: Name of the file
+
+    Returns:
+        Dictionary with node and its corresponding neighbors
+    """
+    node_dict = {}
+    node_dict["edge"] = False
+    node_dict["name"] = os.path.splitext(filename)[0].split("/")[-1]
+    node_dict["neighbors"] = []
+    if file_data and file_data.get("ethernet_interfaces"):
+        if type(file_data["ethernet_interfaces"]) == dict:
+            for key, value in file_data["ethernet_interfaces"].items():
+                if "peer_interface" not in value:
+                    continue
+                neighbor_dict = {}
+                neighbor_dict["neighborDevice"] = value["peer"]
```

Comment:
please use `neighbor_device` rather than `neighborDevice` to be consistent in case. Same for other keys